### PR TITLE
Add Python 3.12 Support

### PR DESCRIPTION
**Pull request by damaestro**
- Remove use of imp
 - Update tests for unittest changes
 - Update ConfigParser read

### CODE REVIEW

**Comment by deniszh** [comment]
Hi @damaestro 

Could you please rebase against latest changes in master? I fixed the tests, should be green now.

File: lib/carbon/routers.py
Comment (damaestro) [inline lib/carbon/routers.py:1]: The porting to importlib is incomplete and needs to be reviewed by a maintainer that knows how this should behave.

imp | importlib
-- | --
imp.get_suffixes() | importlib.machinery.SOURCE_SUFFIXES, importlib.machinery.EXTENSION_SUFFIXES, and importlib.machinery.BYTECODE_SUFFIXES
imp.load_module() | importlib.import_module()

https://docs.python.org/dev/whatsnew/3.12.html#removed

The function signature has changed and the call needs to be cleaned up: https://docs.python.org/dev/library/importlib.html#importlib.import_module

Comment (damaestro) [reply lib/carbon/routers.py:1]: Use of `imp` was only for `carbon-client` and that was removed https://github.com/graphite-project/carbon/commit/1ebd4c16cf62bcf6630d77ba593f3f937d278951

I've removed the function related to this functionality.

**Comment by deniszh** [comment]
Didn't trigger tests for some reason. I redid tests in #951, LGTM, will merge.

**Comment by deniszh** [comment]
Thank you, @damaestro, merged with master.

**Pull request closed without merging**